update Version-specific update instructions (#2135)
meilisearch/documentation: learn/advanced/updating.md

If you're updating your Meilisearch instance on cloud platforms like DigitalOcean, AWS, or GCP, ensure that you can connect to your cloud instance via SSH. Depending on the user you are connecting with (root, admin, etc.), you may need to prefix some commands with `sudo`.

> [Refer to the Version-specific update warnings sections for more details on upgrading from specific versions.](#version-specific-update-warnings)

Reviewer: ```suggestion
If migrating to the latest version of Meilisearch will cause you to skip multiple versions, this may require changes to your codebase. [Refer to our version-specific update warnings for more details](#version-specific-warnings).
```

<CodeSamples id="updating_guide_check_version_old_authorization_header" />

> - If you are updating from **v0.25 or below**:

Reviewer: ```suggestion
- If you are updating from **v0.25 or below**, be aware that:
```

}
```

::: note

> If you get the `missing_authorization_header` error, you might be using **v0.24 or below**. Use the `X-Meili-API-Key: API_KEY` authorization header:

Author: I did add ` # -H 'X-Meili-API-Key: API_KEY' for v0.24 or below` to relavent code samples

If everything looks good, then congratulations! You successfully migrated your database to the latest version of Meilisearch. Be sure to check out the [changelogs](https://github.com/meilisearch/MeiliSearch/releases).

> If something went wrong, you can always roll back to the previous version and try again. Be sure to check out the [version-specific update warnings](#version-specific-update-warnings), and feel free to [reach out for help](https://discord.gg/meilisearch) if the problem continues.

Reviewer: ```suggestion
If something went wrong, you can always roll back to the previous version. Feel free to [reach out for help](https://discord.gg/meilisearch) if the problem continues. If you successfully migrated your database but are having problems with your codebase, be sure to check out our [version-specific warnings](#version-specific-warnings).
```

:::

::::

## Version-specific update instructions

This section contains instructions for upgrading from specific versions. Most versions don't require version-specific steps and you should be able to upgrade directly. If the version you are upgrading from isn't listed here, no additional steps are required.

- If you are updating from **v0.25 or below**, be aware that:
  - The `private` and `public` keys have been deprecated and replaced by two default API keys with similar permissions: `Default Admin API Key` and `Default Search API Key`.
  - The `updates` API has been replaced with the `tasks` API.

> - If you are **updating to v0.28 or above**, existing keys will have their `key` and `uid` fields regenerated.

Author: This applies to all users[ updating from v0.27 and below](https://meilisearch.slack.com/archives/C010VV3SHQF/p1675329315659259?thread_ts=1675325777.472189&cid=C010VV3SHQF)

> - If you are **updating from v0.27 or above**, existing keys will have their `key` and `uid` fields regenerated.

Author: ```suggestion
- If you are **updating from v0.27 or below**, existing keys will have their `key` and `uid` fields regenerated.
```

:::

::::

> ## Version-specific update warnings

Reviewer: ```suggestion
## Version-specific warnings
```

> This section contains instructions for upgrading from specific versions. Most versions don't require version-specific steps and you should be able to upgrade directly. If the version you are upgrading from isn't listed here, no additional steps are required.

Reviewer: ```suggestion
After migrating to the most recent version of Meilisearch, your code-base may require some changes. This section contains warnings for some of the most impactful version-specific changes. For full changelogs, see the [releases tab on GitHub](https://github.com/meilisearch/meilisearch/releases).
```
